yt_dlp/options.py

```diff
@@ -52,7 +53,7 @@ def _scrub_eq(o):
 def parseOpts(overrideArguments=None):
     def _readOptions(filename_bytes, default=[]):
         try:
-            optionf = open(filename_bytes)
+            optionf = io.open(filename_bytes, 'r', encoding='utf8')
```

Comment:
No more than 10 lines later, we have this:

```
            if sys.version_info < (3,):
                contents = contents.decode(preferredencoding())
```

With Python 2 no longer supported, this is dead code anyway, but it is still an inconsistency.

Reply:
I guess one might as well rewrite this whole method to use the `with` statement and catch `FileNotFoundError` only, instead of a general `IOError`.

Reply:
> ...rewrite this whole method to use the `with` statement...

Fair enough, feel free to suggest the full python code, otherwise I'll certainly consider it for an additional PR.

But now that you mention it, apparently the original intention was for `preferredencoding()` to be used (whatever that is) for reading the options file. The problem then is that python 3 never makes it this far because the `UnicdoeDecodeError` exception is not caught in the first try block. So a completely different approach, more in the spirit of the original code, would be to read the file ***in binary mode in the first place***, and then proceed to explicitly decode the `bytes` using `preferredencoding()`.